### PR TITLE
MUL-727: Refactoring.

### PR DESCRIPTION
**Pull request by Enmk**
Removed transaction_has_trait() API;
Moved BinaryData into binary_data.h/cpp;
Moved validate_address() to blockchain.h/cpp;
Added bitcoin.h, ethereum.h, golos.h that have blockchain-specific enums;
Split utility into error_utility and binary_data_utility;
A bit simplified error management.

### CODE REVIEW

File: multy_core/src/api/binary_data.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+namespace
+{
+using namespace multy_core::internal;
+} // namespace
+
+Error* make_binary_data_clone(const BinaryData* source, BinaryData** new_binary_data)
+{
+    ARG_CHECK(source);
+    return make_binary_data_from_bytes(
```
Comment (PashaKlybik) [inline multy_core/src/api/binary_data.cpp:24]: Add check new_binary_data to and down

**Review by PashaKlybik — COMMENTED** [review]
Pleas fix it. The rest all ok

File: multy_core/src/api/binary_data.cpp (hunk 1 above)
Comment (Enmk) [reply multy_core/src/api/binary_data.cpp:24]: It is checked in `make_binary_data_from_bytes()`

**Comment by Enmk** [comment]
Fixed

**Review by PashaKlybik — APPROVED** [review]
Good!

**Pull request merged**